### ISSUE TITLE
Removes parentheses when "and" is combined with ternary expression

### ISSUE THREAD

**Issue by ZeeCoder** [prettier/plugin-php#1245]
**@prettier/plugin-php v0.11.2**
[Playground link](https://loilo.github.io/prettier-php-playground/#N4IgDgTgpgLjCWUIHkwIPYDsDOIBcok8mMA6vACYwAW+AHAAwA0IMAhgEblW14AsLAK7YoAFU648AMzYAbES2zEA5rKgBFQehhR8M+VBYwIbeLJUBhdAFtrbAAoAJe-hCYsulhxMBjKAGUYAE81VzBsCABaACYQFmgAR0F4aHsTZTs9OQUQYhEIGDS2DLYsgwBfFigKeBh0FDR4LElQNmwYMpFK3MwwQQ68EAAeAH4wajAAHUxpgBJsGygANSQglYgggBksZSW5QSgARgACAF5jw4BuOYXrZdX1rZ292QPos+Po65nMADc2CAAfQogmsYAAFNNjtDjuD5otHo9tphdvsjsc2JgKMd4XdEatkajXlBogBKKEwykjY4AcmMB2O-2JNIplOheFp+hEjLRLMwpO+IHKQA)

**Input:**
```php
<?php

$someVeryVeryLongValue1 = 1;
$someVeryVeryLongValue2 = 2;

var_dump(
    ($someVeryVeryLongValue1 and $someVeryVeryLongValue2)
        ? 'true value'
        : 'false value'
);

```

**Output:**
```php
<?php

$someVeryVeryLongValue1 = 1;
$someVeryVeryLongValue2 = 2;

var_dump(
    $someVeryVeryLongValue1 and $someVeryVeryLongValue2
        ? 'true value'
        : 'false value'
);

```

The above two are not equal in execution.
It would be fine to do this for the && operator, as that has a higher precedence than the ternary, but the same is not true for "and":
https://www.php.net/manual/en/language.operators.precedence.php

**Comment by evilebottnawi** [comment]
PR welcome

**Comment by ZeeCoder** [comment]
Any starting points you would recommend?

**Comment by evilebottnawi** [comment]
Here logic for parens https://github.com/prettier/plugin-php/blob/master/src/needs-parens.js

**Comment by ZeeCoder** [comment]
@czosel @evilebottnawi this is still an issue in v12.

**Comment by czosel** [comment]
Strange. I’ll try to look into this in the coming days.

**Comment by czosel** [comment]
Ah, the answer is simple: The fix is not part of the 0.12.0 release - it should be working in master though.

**Comment by ZeeCoder** [comment]
All right, guess I'll just wait for the next release and re-test then.
